webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,235 @@
+  onLaunch?: () => void;
+  visible?: boolean;
+}
+
+const NotebookModal: React.FC<Props> = (
+  { visible = false, onLaunch, ...props }: Props,
+) => {
+  const storage = useStorage(STORAGE_PATH);
+  const [ showFullConfig, setShowFullConfig ] = useState(false);
+  const [ templates, setTemplates ] = useState<Template[]>([]);
+  const [ resourcePools, setResourcePools ] = useState<ResourcePool[]>([]);
+  const [ showResourceType, setShowResourceType ] = useState(false);
+  const [ resourceType, setResourceType ] = useState<ResourceType | undefined>
+  (storage.getWithDefault(STORAGE_KEY, { type: undefined }).type);
+  const [ form ] = Form.useForm();
+
+  const fetchTemplates = useCallback(async () => {
+    try {
+      setTemplates(await getTaskTemplates({}));
+    } catch {}
+  },[]);
+
+  useEffect(() => {
+    fetchTemplates();
+  }, [ fetchTemplates ]);
+
+  const fetchResourcePools = useCallback(async () => {
+    try {
+      setResourcePools( await getResourcePools({}));
+    } catch {}
+  }, []);
+
+  useEffect(() => {
+    fetchResourcePools();
+  }, [ fetchResourcePools ]);
+
+  const fetchConfig = useCallback(async () => {
+    try {
+      const values: NotebookConfig = form.getFieldsValue(true);
+      const config = await previewNotebook(
+        values.type === ResourceType.CPU ? 0 : values.slots,
+        values.template === 'default'? undefined : values.template,
+        values.name,
+        values.pool,
+      );
+      form.setFieldsValue({ config: yaml.dump(config) });
+    } catch {}
+  }, [ form ]);
+
+  useEffect(()=> {
+    if (showFullConfig) fetchConfig();
+  }, [ showFullConfig, fetchConfig ]);
+
+  const storeConfig = useCallback((_, values: NotebookConfig) => {
+    delete values.name;
+    storage.set(STORAGE_KEY,values);
+  }, [ storage ]);
+
+  const handleSecondary = useCallback(async () => {
+    if (showFullConfig) {
+      setShowFullConfig(false);
+    } else {
+      try {
+        await form.validateFields();
+        setShowFullConfig(true);
+      } catch (e) {}
+    }
+  },[ form, showFullConfig ]);
+
+  const handleCreateEnvironment = useCallback(
+    (values) =>{
+      if (showFullConfig) {
+        launchNotebook(yaml.load(form.getFieldValue('config')) as RawJson);
+      } else {
+        launchNotebook(
+          undefined,
+          resourceType === ResourceType.CPU ? 0 : values.slots,
+          values.template === 'default' ? undefined : values.template,
+          values.name,
+          values.pool,
+        );
+      }
+      if (onLaunch) onLaunch();
+    },
+    [ showFullConfig, form, onLaunch, resourceType ],
+  );
+
+  const handleResourcePoolUpdate = useCallback((selectedPoolName: string) => {
+    const selectedPool = resourcePools.find(pool => pool.name === selectedPoolName);
+    if (!selectedPool) return;
+    const hasCPUCapacity = selectedPool.cpuContainerCapacityPerAgent > 0;
+    const hasGPUCapacity = selectedPool.slotsPerAgent && selectedPool.slotsPerAgent > 0;
```

Comment:
This should be `slotsAvailable > 0` (for static agents) or `slotsPerAgent > 0` (for dynamic agents).

Reply:
How do I know if it is a static vs dynamic agent? Also, what is the behavior if someone tries to use more slots than are available?

Reply:
Actually, you don't need to know if the pool has static agents or dynamic agents. If `slotsAvailable > 0`, it means agents of more than one slots are connected. `slotsPerAgent > 0` means dynamic agents of more than one slots are configured. `slotsAvailable > 0` alone cannot determine if there might be more dynamic agents of more than one slots connected. So `slotsAvailable > 0 or slotsPerAgent > 0` would be the expression to determine if there exist GPU capacity.